frontend/src/editor/demoeditor.tsx

```diff
@@ -0,0 +1,145 @@
+import { useRef, useState, useEffect } from 'react';
+import ReactDOM from 'react-dom';
+import { Auth0ContextInterface } from '@auth0/auth0-react';
+import * as SidebarInner from '@/pages/app';
+import LexicalEditor from './editor';
+import classes from './demo.module.css';
+
+const MAX_WORD_COUNT = 200;
+
+function Sidebar({ editorAPI }: { editorAPI: EditorAPI }) {
+	return (
+		<SidebarInner.default editorAPI={ editorAPI }/>
+	);
+}
+
+function App() {
+	// This is a reference to the current document context
+	const docContextRef = useRef<DocContext>({
+		beforeCursor: '',
+		selectedText: '',
+		afterCursor: ''
+	});
+
+	// Add word count state
+	const [wordCount, setWordCount] = useState<number>(0);
+	// Add state to track when word count exceeds limit
+	const [isOverLimit, setIsOverLimit] = useState<boolean>(false);
+
+	// Since this is a list, a useState would have worked as well
+	const selectionChangeHandlers = useRef<(() => void)[]>([]);
+
+	const handleSelectionChange = () => {
+		selectionChangeHandlers.current.forEach(handler => handler());
+	};
+
+	// Add effect to prevent input when word count exceeds limit
+	useEffect(() => {
+		const editorElement = document.querySelector(`.${classes.editor}`);
+		if (!editorElement) return;
+
+		// Function to handle input events
+		const handleBeforeInput = (event: Event) => {
+			const inputEvent = event as InputEvent;
+			if (isOverLimit && (
+				inputEvent.inputType === 'insertText' ||
+				inputEvent.inputType === 'insertFromPaste' ||
+				inputEvent.inputType.startsWith('insert')
+			)) {
+				event.preventDefault();
+			}
+		};
+
+		// Add event listener
+		editorElement.addEventListener('beforeinput', handleBeforeInput);
+
+		// Update visual feedback when over limit
+		if (isOverLimit) {
+			editorElement.classList.add(classes.overLimit);
+		}
+		else {
+			editorElement.classList.remove(classes.overLimit);
+		}
+
+		// Clean up when component unmounts
+		return () => {
+			editorElement.removeEventListener('beforeinput', handleBeforeInput);
+		};
+	}, [isOverLimit]);
+
+	const editorAPI: EditorAPI = {
```

Comment:
This is redundant with the non-demo editorAPI implementation. Let's pull that code into a separate module and import it in both places.